src/OrchardCore/OrchardCore.DisplayManagement/ModelBinding/LocalModelBinderAccessor.cs

```diff
@@ -1,15 +1,21 @@
-﻿using System.Threading;
+using Microsoft.AspNetCore.Http;
 
 namespace OrchardCore.DisplayManagement.ModelBinding
 {
     public class LocalModelBinderAccessor : IUpdateModelAccessor
     {
-        private readonly AsyncLocal<IUpdateModel> _storage = new AsyncLocal<IUpdateModel>();
+        private readonly static object Key = typeof(LocalModelBinderAccessor);
+        private readonly HttpContext _httpContext;
+
+        public LocalModelBinderAccessor(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContext = httpContextAccessor.HttpContext;
+        }
 
         public IUpdateModel ModelUpdater
         {
-            get { return _storage.Value; }
```

Comment:
Why does it work better?

Reply:
Here i copied the comment i did in #1691

When we store an updater through the `LocalModelBinderAccessor` which uses an `AsyncLocal` value, we may be already in a switched thread so the value may not survive other thread switching.

It is not the same with the `HttpContextFactory` which creates an `HttpContext` and stores it through a registered `IHttpContextAccessor`, **because it does this earlier before processing the request**.

For testing, storing the updater through the actual `HttpContext` fixes the issue. I will suggest a PR for this.